nx/source/services/hid.c

```diff
@@ -227,6 +230,32 @@ void hidScanInput(void) {
 
         g_controllerDown[i] = (~g_controllerOld[i]) & g_controllerHeld[i];
         g_controllerUp[i] = g_controllerOld[i] & (~g_controllerHeld[i]);
+
+        u32 type = g_controllerHeaders[i].type;
```

Comment:
Only run this when SixAxis is activated.

Reply:
How would you suggest I go about detecting that? I can add a check to not do the memcpy when the SixAxis is deactivated but the rest of the code can't easily do that without bookkeeping that could get out of sync.

If the former then I should also add cleanup when stopping a SixAxis sensor to make sure you don't read out old states.

Reply:
Set a flag in a table for the specified controller in the enable/disable funcs.